components/director/internal/domain/labeldef/service_test.go

```diff
@@ -140,19 +141,267 @@ func TestServiceList(t *testing.T) {
+			Key:    key,
+			Tenant: tenant,
+			Schema: newSchema,
+		}
+
+		existingLabels := []*model.Label{
+			{
+				ID:     "b9566e9d-83a2-4091-8c65-7a512b88f89e",
+				Tenant: tenant,
+				Key:    key,
+				Value: map[string]interface{}{
+					"firstName": "val",
+					"lastName":  "val2",
+					"age":       1235,
+				},
+				ObjectID:   "foo",
+				ObjectType: model.RuntimeLabelableObject,
+			},
+			{
+				ID:     "2037fc3d-be6c-4489-94cf-05518bac709f",
+				Tenant: tenant,
+				Key:    key,
+				Value: map[string]interface{}{
+					"firstName": "val3",
+					"lastName":  "val4",
+				},
+				ObjectID:   "bar",
+				ObjectType: model.ApplicationLabelableObject,
+			},
+		}
+
+		defWithID := in
+		defWithID.ID = fixUUID()
+
+		mockRepository.On("GetByKey", mock.Anything, tenant, key).Return(&ld, nil).Once()
+		mockRepository.On("Update", mock.Anything, defWithID).Return(nil)
+
+		mockLabelRepository.On("ListByKey", context.TODO(), tenant, key).Return(existingLabels, nil).Once()
+
+		ctx := context.TODO()
+		sut := labeldef.NewService(mockRepository, mockLabelRepository, nil)
+		// WHEN
+		err := sut.Update(ctx, in)
+		// THEN
+		require.NoError(t, err)
+	})
+
+	t.Run("returns error when existing label doesn't match new schema", func(t *testing.T) {
+		// GIVEN
+		mockRepository := &automock.Repository{}
+		defer mockRepository.AssertExpectations(t)
+
+		mockLabelRepository := &automock.LabelRepository{}
+		defer mockLabelRepository.AssertExpectations(t)
+
+		oldProperty := "oldProperty"
+		nonExistingProperty := "nonExistingProp"
+
+		ld := model.LabelDefinition{
+			ID:     fixUUID(),
+			Tenant: tenant,
+			Key:    key,
+			Schema: fixSchema(t, oldProperty, "string", "desc", "oldProperty"),
+		}
+
+		in := model.LabelDefinition{
+			ID:     fixUUID(),
+			Key:    key,
+			Tenant: tenant,
+			Schema: fixSchema(t, nonExistingProperty, "integer", "desc", "nonExistingProp"),
+		}
+
+		existingLabels := []*model.Label{
+			{
+				ID:     "b9566e9d-83a2-4091-8c65-7a512b88f89e",
+				Tenant: tenant,
+				Key:    oldProperty,
+				Value: map[string]interface{}{
+					"key": "val",
+				},
+				ObjectID:   "foo",
+				ObjectType: model.RuntimeLabelableObject,
+			},
+			{
+				ID:     "2037fc3d-be6c-4489-94cf-05518bac709f",
+				Tenant: tenant,
+				Key:    oldProperty,
+				Value: map[string]interface{}{
+					"key": "val2",
+				},
+				ObjectID:   "bar",
+				ObjectType: model.ApplicationLabelableObject,
+			},
+		}
+
+		defWithID := in
+		defWithID.ID = fixUUID()
+
+		mockRepository.On("GetByKey", mock.Anything, tenant, key).Return(&ld, nil).Once()
+
+		mockLabelRepository.On("ListByKey", context.TODO(), tenant, key).Return(existingLabels, nil).Once()
+
+		ctx := context.TODO()
+		sut := labeldef.NewService(mockRepository, mockLabelRepository, nil)
+		// WHEN
+		err := sut.Update(ctx, in)
+		// THEN
+		require.Error(t, err)
+		require.EqualError(t, err, "label with key oldProperty is not valid against new schema for Runtime with ID foo")
```

Comment:
minor: add qoutes for 'oldProperty` and `foo`

Reply:
done